packages/syft/src/syft/service/action/action_store.py

```diff
@@ -91,8 +92,9 @@ def get_pointer(
             if uid in self.data:
                 obj = self.data[uid]
                 if isinstance(obj, TwinObject):
-                    obj = obj.mock
-                    obj.syft_twin_type = TwinMode.MOCK
+                    obj = (
+                        obj.mock if not is_action_data_empty(obj.mock) else obj.private
+                    )
```

Comment:
@shubham3121 this part makes it so that guest client gets back a private pointer if mock is empty